### ISSUE TITLE
Vue CLI vs. Angular CLI

### ISSUE THREAD

**Issue by kreinholz** [susanBuck/e28-fall19#87]
Reviewing the materials for week 8, I was struck by just how similar Vue CLI and the Vue Development Server are to Angular CLI and the Angular Development Server as taught in CSCI E-31.

I guess my question is, if we’re taking the comprehensive approach, is there much incentive to learn Vue vice focusing on Angular? I realize this is a _very_ loaded question, but just looking at market share, Angular seems like the more in-demand framework.

If being used with the integrated approach, then I see a very obvious advantage to Vue over Angular (aside from being able to write code in pure JavaScript and not TypeScript), but with the comprehensive approach, building the entire application from the ground up, why Vue and not Angular (or React)?

Granted, Vue gives the flexibility to put the HTML, CSS, and JS for a single component into a single file, whereas Angular forces you to separate technologies, instead housing each component within a directory, but seeing Vue CLI in action, and the Vue Development Server, I was struck by just how similar Vue is to Angular! The end result, compiling a SPA into a dist folder, likewise is just what I did with Angular.

Don’t get me wrong, I enjoy writing code for Vue, and think the flexibility of the integrated approach is a killer feature. I was just a bit surprised by how much the comprehensive approach resembles Angular development.

**Comment by susanBuck** [comment]
These are all interesting points/questions! 

I think you'll find <https://vuejs.org/v2/guide/comparison.html> an interesting read - it compares Vue vs React, Angular (1 & 2), and a few other frameworks.

tl;dr:
+ Angular requires TypeScript, Vuejs makes it optional (this is either a pro/con depending on how you feel about TypeScript)
+ Performance difference is negligible
+ Vue is smaller
+ Vue is more flexible
+ Vue is easier to learn (which isn't necessarily a pro if you're coming from Angular and already know it :)

We'll spend some time later in the semester taking a closer look at Vue.js vs. other frameworks - but curious to hear the thoughts of classmates who have experience with other frameworks. What have you liked/disliked about Vue so far compared to those other frameworks?

Keep in mind - I know this is a Vue.js class - but I'm not at all in any sort of "Vue is best!" mindset, so no one should worry about expressing vue-negative opinions. I very rarely take that stance that any language/technology is "best" just because there are often pros and cons to different systems and lots of factors to consider (team preference, experience, goals, etc.).

**Comment by phildarnowsky** [comment]
Speaking to the question of market share, as someone who's interviewed a fair number of engineers in the past, I wouldn't worry too much about the exact technology you are learning. We use React and Typescript (which BTW I recommend highly), and I recently recommended a candidate with zero production experience in either, because in other respects they were a strong candidate. I believe most managers of engineers would agree that a candidate without experience in a particular technology, but who is eager to learn, can still be a candidate worth looking at.
